cypress/support/fragments/users/financialTransactionDetailReportModal.js

```diff
@@ -185,4 +185,31 @@ export default {
   verifyFeeFineOwnerIsRequiredErrorMessage() {
     cy.expect(financialReport.find(HTML(including('"Fee/fine owner" is required'))).exists());
   },
+
+  verifyFileExists(fileName) {
+    cy.readFile(`cypress/downloads/${fileName}`);
+  },
+
+  checkDownloadedFile(fileName, action, actionAmount, rowNumber, data) {
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(3000); // wait for the file to load
+    cy.readFile(`cypress/downloads/${fileName}`).then((fileContent) => {
+      // Split the contents of a file into lines
+      const fileRows = fileContent.split('\n');
+
+      expect(fileRows[0].trim()).to.equal(
+        '"Fee/fine owner","Fee/fine type","Fee/fine billed amount","Fee/fine billed date/time","Fee/fine created at","Fee/fine source","Fee/fine details","Action","Action amount","Action date/time","Action created at","Action source","Action status","Action additional staff information","Action additional patron information","Payment method","Payment transaction information","Waive reason","Refund reason","Transfer account","Patron name","Patron barcode","Patron group","Patron email address","Instance","Contributors","Item barcode","Call number","Effective location","Loan date/time","Due date/time","Return date/time","Loan policy","Overdue policy","Lost item policy","Loan details"',
```

Comment:
Maybe better that string add to const and after that explore this in the method?

Reply:
Fixed